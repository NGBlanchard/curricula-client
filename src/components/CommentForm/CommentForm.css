.CommentForm {
  margin-bottom: 50px;
}