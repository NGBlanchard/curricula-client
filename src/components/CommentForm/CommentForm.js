import React, { Component } from 'react'
import CourseContext from '../../context/CourseContext'
import TokenService from '../../services/token-service';
import CourseApiService from '../../services/course-api-service'
import { Button, Textarea } from '../Utils/Utils'
import './CommentForm.css'



export default class CommentForm extends Component {
  state = {
    user: "",
  }
  static contextType = CourseContext

    componentDidMount = () => {
      this.context.clearError()
      const userId = TokenService.getUserId()
      this.setState({
      user: userId
      })
    }

  

  onBlurb = e => {
    e.preventDefault()
    const { course } = this.context
    const { content } = e.target
    const comment = {
      content: content.value,
      date_created: new Date(),
      course_id: course.id,
      user_id: this.state.user
    }
    CourseApiService.postComment(comment)
      .then(this.context.addComment)
      .then(() => {
        content.value = ''
      })
      .catch(this.context.setError)
  }

  render() {
    return (
      <form
        className='CommentForm'
        onSubmit={this.onBlurb}
      >
        <div className='content'>
          <Textarea
            required
            aria-label="Here's to the Alcove..."
            name='content'
            id='content'
            cols='30'
            rows='3'
            placeholder="Here's to the Alcove...">
          </Textarea>
        </div>
        <Button type='submit'>
          Blurb This Course
        </Button>
      </form>
    )
  }
}
