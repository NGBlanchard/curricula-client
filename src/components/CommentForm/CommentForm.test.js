import CommentForm from "./CommentForm";
import React from "react";
import { configure, mount } from "enzyme";
import toJson from "enzyme-to-json";

import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("<CommentForm />", () => {
  const comment = {
    id: 1,
    user_id: 2
  };

  const context = {
    clearError() {
      this.setState({ error: null });
    }
  };
  it("Renders without crashing", () => {
    mount(
      <CommentForm key={comment.id} comment={comment} user={comment.user_id} />,
      context
    );
  });
  it("renders the form by default", () => {
    const wrapper = mount(
      <CommentForm key={comment.id} comment={comment} user={comment.user_id} />,
      context
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
