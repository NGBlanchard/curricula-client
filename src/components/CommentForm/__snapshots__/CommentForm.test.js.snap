// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentForm /> renders the form by default 1`] = `
<CommentForm
  comment={
    Object {
      "id": 1,
      "user_id": 2,
    }
  }
  user={2}
>
  <form
    className="CommentForm"
    onSubmit={[Function]}
  >
    <div
      className="content"
    >
      <Textarea
        aria-label="Here's to the Alcove..."
        cols="30"
        id="content"
        name="content"
        placeholder="Your blurb here..."
        required={true}
        rows="3"
      >
        <textarea
          aria-label="Here's to the Alcove..."
          className="Textarea "
          cols="30"
          id="content"
          name="content"
          placeholder="Your blurb here..."
          required={true}
          rows="3"
        />
      </Textarea>
    </div>
    <Button
      type="submit"
    >
      <button
        className="Button "
        type="submit"
      >
        Blurb This Course
      </button>
    </Button>
  </form>
</CommentForm>
`;
